src/platform/Linux/ThreadStackManagerImpl.cpp

```diff
@@ -174,6 +174,19 @@ CHIP_ERROR ThreadStackManagerImpl::_SetThreadProvision(const Internal::DeviceNet
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ThreadStackManagerImpl::_SetThreadProvision(const uint8_t * operationalDataset, uint32_t operationalDatasetLen)
+{
+    mOperationalDatasetTlv = std::vector<uint8_t>(operationalDataset, operationalDataset + operationalDatasetLen);
+
+    // post an event alerting other subsystems about change in provisioning state
+    ChipDeviceEvent event;
+    event.Type                                           = DeviceEventType::kServiceProvisioningChange;
```

Comment:
Shall we post `ServiceProvisionChange` here or `ThreadStateChange` is a better fit?